http.go

```diff
@@ -170,6 +174,13 @@ func probeHTTP(target string, w http.ResponseWriter, module Module) (success boo
 			success = true
 		}
 
+		// If the response was gzipped and was transparently uncompressed
+		if resp.Uncompressed {
+			content_compressed = 1
+			if body, err := ioutil.ReadAll(resp.Body); err == nil {
+				content_length = int64(len(body))
```

Comment:
Agree as well. However, the net/http client actually discards the Content-Length header when it has set accept gzip. There are basically two possibilities that I can see:
1. Manage gunzipping the body in blackbox_exporter if `Content-Encoding: gzip` is set (note that this is also thrown away by net/http if they set the accept header)
2. Try to estimate the original length by gzipping it ourselves. This is quite a hack, and won't give correct results. 

As for the new metric, `probe_http_content_decompressed_length`?